### DISCUSSION TITLE
slow rendering time

### DISCUSSION THREAD

**Discussion by lykius** [NVlabs/instant-ngp#271] [Q&A]
Hi!

Instant-NGP is great, thanks for sharing!
I'm successfully training it in "nerf" mode, using the python binding provided by you in `scripts/run.py`.
However, when I use the same code to render 500 images (1440x1080) it takes almost 10 minutes on a 3090 gpu. I disabled the metrics computation to be sure to focus only on rendering.

Is it the expected rendering speed?
Thanks in advance!

**Comment by Tom94** [comment]
By default, `scripts/run.py` renders with 8 samples per pixel (effectively doing 8x superresolution for anti aliasing). You can reduce this to 1 in the script (`spp` variable) to get 8x faster rendering for ~2 minutes in total.

Another aspect is that saving the images means encoding them on the CPU. On my machine, this is slower than the actual rendering part. You might benefit from multi-threading that through Python.

(Speed-of-light, my 3090 needs ~100ms for a 1440x1080 frame, so the best case for 500 images would be 50 seconds there.)

Cheers!

**Comment by lykius** [comment]
That helps a lot, thanks!!
